tutorial/lesson_21_auto_scheduler_generate.cpp

```diff
@@ -0,0 +1,190 @@
+// Halide tutorial lesson 21: Auto-Scheduler
+
+// This lesson demonstrates how to use the auto-scheduler to generate a
+// copy-pastable CPU schedule that can be subsequently improved upon.
+
+// On linux or os x, you can compile and run it like so:
+
+// g++ lesson_21_auto_scheduler_generate.cpp ../tools/GenGen.cpp -g -std=c++11 -fno-rtti -I ../include -L ../bin -lHalide -lpthread -ldl -o lesson_21_generate
+// export LD_LIBRARY_PATH=../bin   # For linux
+// export DYLD_LIBRARY_PATH=../bin # For OS X
+// ./lesson_21_generate -o . -f conv_layer target=host
+// g++ lesson_21_auto_scheduler_run.cpp brighten_*.o -ldl -lpthread -o lesson_21_run
+// ./lesson_21_run
+
+// If you have the entire Halide source tree, you can also build it by
+// running:
+//    make tutorial_lesson_21_auto_scheduler_run
+// in a shell with the current directory at the top of the halide
+// source tree.
+
+#include "Halide.h"
+#include <stdio.h>
+
+using namespace Halide;
+
+// We will define a generator to auto-schedule.
+class AutoScheduleGen : public Halide::Generator<AutoScheduleGen> {
```

Comment:
"Gen" as an abbreviation for Generator is something we don't anywhere else in source code (RunGen/GenGen notwithstanding), and I'm reluctant to start now. How about "AutoScheduled" instead?